### DISCUSSION TITLE
Failed to connect to proxy URL: "http://socks5://"

### DISCUSSION THREAD

**Discussion by uchiha-pain** [aws/aws-cli#7248] [Q&A] [answered]
I have a fully private HSCN-VPC with 2 public and 2 private subnets. This HSCN-VPC is peered with another internet-VPC. Now I am deploying a fully private eks cluster to the private subnet of the HSCN-VPC. I have created a ubuntu machine in the private subnet of the HSCN-VPC and trying to deploy a fully private cluster. Cluster is successfully deployed, but the problem is I am not able to run any of the kubectl, eks, or aws commands even though i am trying to run these commands from the machine inside the HSCN-VPC. I have installed the [aws-iam-authenticator](https://docs.aws.amazon.com/eks/latest/userguide/install-aws-iam-authenticator.html), and now trying to [update kubeconfig](https://docs.aws.amazon.com/eks/latest/userguide/create-kubeconfig.html) in order to run kubectl commands. When I ran this command
`aws sts get-caller-identity`
I got the following error
`Failed to connect to proxy URL: "http://socks5://192.168.0.69:19000"`
Any idea how can I fix it or run the kubectl and eks commands for my private cluster even though I am already inside the private subnet of the private HSCN-VPC?
Thanks

**Comment by tim-finnigan** [comment, marked as answer]
Hi @uchiha-pain thanks for reaching out. I'll link a couple related issues: https://github.com/aws/aws-cli/issues/6570 and https://github.com/aws/aws-cli/issues/5976. There is a currently an open issue in the botocore repository tracking the request to support SOCKS5 proxies: https://github.com/boto/botocore/issues/2540. We recommend adding a 👍 to that issue to indicate your support and if you have additional details to share regarding your use case you can also provide those there.
